bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/obs/plannedtime/PlannedTime.java

```diff
@@ -223,17 +222,30 @@ public static CategorizedTimeGroup apply(CategorizedTime... col) {
             return times.hashCode();
         }
 
-        public Map<Category, CategorizedTime> maxTimes() {
+        public Pair<Map<Category, CategorizedTime>, Map<Category, ImList<CategorizedTime>>> maxTimes() {
             Map<Category, CategorizedTime> max = new TreeMap<>();
+            Map<Category, ImList<CategorizedTime>> losers = new TreeMap<>();
+
             for (CategorizedTime ct : times) {
                 CategorizedTime cur = max.get(ct.category);
-                if ((cur == null) || (cur.compareTo(ct) < 0)) {
+
+                ImList<CategorizedTime> currLosers = losers.get(ct.category);
+                if (currLosers == null) {
+                    losers.put(ct.category, ImCollections.emptyList());
+                }
+
+                if (cur == null) {
                     max.put(ct.category, ct);
+                } else if (cur.compareTo(ct) < 0) {
+                    max.put(ct.category, ct);
+                    losers.put(ct.category, currLosers.append(cur));
+                }
+                else {
+                    losers.put(ct.category, currLosers.append(ct));
```

Comment:
This is so dirty because I was missing some method in `TreeMap` to *mappend* values when the key is present, I guess there is a better way but didn't find it. Also I don't know how to handle those null cases better.

Reply:
i don't think there is a way around on `TreeMap`. Possibly `ConcurrentMap` has a `putIfPresent/Missing` or something like that